From 0b35dcfabdd2eefbf1c40ce25fd071a26409c53e Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Tue, 27 Nov 2018 08:53:10 +0100
Subject: [PATCH 12/15] Fix issue #4: 'prefix' is too short.

Contents like "A100:" is too long to be NULL-terminated, so accessing
the 'prefix' buffer bombs. Add a sixth byte for the NULL terminator.
---
 PROTOTYPES/gb_gates.ch | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/PROTOTYPES/gb_gates.ch b/PROTOTYPES/gb_gates.ch
index c1e782d..a6c93ec 100644
--- a/PROTOTYPES/gb_gates.ch
+++ b/PROTOTYPES/gb_gates.ch
@@ -55,6 +55,12 @@ static Vertex* new_vert(char t)
   /* the type of the new gate */
 @z
 
+@x l.431
+static char prefix[5]; /* prefix string for vertex names */
+@y
+static char prefix[6]; /* prefix string for vertex names */
+@z
+
 @x l.445
 static Vertex* make2(t,v1,v2)
   char t; /* the type of the new gate */
-- 
2.20.1

